src/main/java/org/reaktivity/rym/internal/RymInstall.java

```diff
@@ -0,0 +1,226 @@
+            boolean resolved = resolveDependencies(options);
+            if (resolved)
+            {
+                logger.info("Dependencies were successfully resolved");
+            }
+            else
+            {
+                logger.error("Dependencies failed to resolve");
+            }
+        }
+        catch (Exception ex)
+        {
+            logger.error(String.format("Error: %s", ex.getMessage()));
+        }
+        finally
+        {
+            logger.sumupProblems();
+        }
+    }
+
+    private void readDepsFile() throws IOException
+    {
+        JsonElement deps = null;
+        try (BufferedReader br = new BufferedReader(new FileReader(DEPENDENCY_FILENAME)))
+        {
+            deps = new JsonParser().parse(br);
+        }
+        // It's possible for a file to pass the parse phase, but still not be valid JSON.
+        // So do an explicit check.
+        if (!deps.isJsonObject())
+        {
+            // TODO Handle file not found
+            throw new JsonSyntaxException(String.format("%s is not in JSON format\n", DEPENDENCY_FILENAME));
+        }
+
+        JsonObject depsObj = (JsonObject)deps;
+
+        JsonElement repositoriesEl = depsObj.get(PROPERTY_REPOSITORIES);
+        if (repositoriesEl != null)
+        {
+            if (!repositoriesEl.isJsonObject())
+            {
+                throw new JsonSyntaxException(String.format("%s is not a JSON object", PROPERTY_REPOSITORIES));
+            }
+            JsonObject repositoriesObj = (JsonObject)repositoriesEl;
+            repositoriesObj.entrySet().forEach(e ->
+            {
+                if (e.getValue().isJsonPrimitive())
+                {
+                    repositories.put(e.getKey(), e.getValue().getAsString());
+                }
+                else
+                {
+                    throw new JsonSyntaxException(
+                        String.format("The value of %s.%s is not a string", PROPERTY_REPOSITORIES, e.getKey()));
+                }
+            });
+        }
+
+        JsonElement dependenciesEl = depsObj.get(PROPERTY_DEPENDENCIES);
+        if (dependenciesEl != null)
+        {
+            if (!dependenciesEl.isJsonObject())
+            {
+                // TODO Handle file not found
+                throw new JsonSyntaxException(String.format("%s is not a JSON object", PROPERTY_DEPENDENCIES));
+            }
+            JsonObject dependenciesObj = (JsonObject)dependenciesEl;
+            dependenciesObj.keySet().forEach(key ->
+            {
+                String artifact = key.indexOf(':') > 0 ? key : String.format("%s:%s", DEFAULT_GROUP_NAME, key);
+                JsonElement versionEl = dependenciesObj.get(key);
+                // TODO Process version here. e.g. Is it a range?
+                String version = versionEl.isJsonPrimitive() ? versionEl.getAsString() : versionEl.toString();
+                dependencies.put(artifact, version);
+            });
```

Comment:
Use `.entrySet()` instead of `.keySet()` to avoid need to call `.get(key)` to get the value for the key.
Also defer injection of default groupId until resolving dependencies, same as treatment of default central repository.

```
dependenciesObj.entrySet().forEach((k, v) -> dependencies.put(k, v.getAsString()));
```